pkg/util/containers/metrics/provider/collector_cache.go

```diff
@@ -106,6 +108,16 @@ func (cc *collectorCache) GetContainerIDForPID(pid int, cacheValidity time.Durat
 	})
 }
 
+// GetContainerIDForInode returns the container ID for given Inode
+// errors are cached as well to avoid hammering underlying collector
+func (cc *collectorCache) GetContainerIDForInode(inode uint64, cacheValidity time.Duration) (string, error) {
```

Comment:
I think we can check that the inode is valid (`if inode <= 2 {return "",}`).
For windows agents or sources that are not running in containers, it might be useful.

Reply:
I would think that we should do the check in the `collectorCache` as to not "split" the logic.

Reply:
I think it's implementation detail and the API should not bother with that. It should return something valid or should not return anything, that's it.